app/presenters/concerns/proposal_details_presenter.rb

```diff
@@ -18,6 +18,13 @@ def fee_related_proposal_details
       end
     end
 
+    def grouped_proposal_details_with_start_numbers
+      grouped_proposal_details.each_with_index.map do |group, index|
+        question_count = grouped_proposal_details.first(index).map(&:last).flatten.count
+        group.dup.push(question_count + 1)
+      end
```

Comment:
Ooh this seems a nice puzzle to solve. Code seems fine and hopefully won't use too much memory if the proposal details we get is very long. Just wanted to ask what the context behind this was?

Reply:
It's so that even though the proposal details are split into separate groups, each with a separate list, the numbering can continue sequentially. This is to make it easier for planners to reference a specific question. They can just say 'question 10', instead of 'group 3, question 4'.